packages/notebook/src/widget.ts

```diff
@@ -1354,16 +1366,29 @@ class Notebook extends StaticNotebook {
    */
   private _evtMouseDown(event: MouseEvent): void {
 
+    const { button, shiftKey } = event;
+
     // We only handle main or secondary button actions.
-    if (!(event.button === 0 || event.button === 2)) {
+    if (!(button === 0 || button === 2)) {
       return;
     }
 
+    // Switch to command mode - let code editor focus determine
+    // a switch to edit mode.
+    this.mode = 'command';
+
     // Shift right-click gives the browser default behavior.
-    if (event.shiftKey && event.button === 2) {
+    if (shiftKey && button === 2) {
       return;
     }
 
+    // Prevent a right click from temporarily focusing the cell and
+    // leaving the notebook in a quasi-state.
+    if (button === 2) {
```

Comment:
This is a pretty big sledgehammer - it says that *any* non-shift right-click event anywhere in the notebook is stopped in its tracks, right? This means that if a custom renderer has its own right-click handler, it's never called. Perhaps we should scope things a little more (e.g., if we're in a cell's input, and the target is in the editor?).